src/lib.rs

```diff
@@ -48,7 +43,7 @@ impl LinkDrop {
     }
 
     /// Create new account and and claim tokens to it.
-    pub fn create_account_and_claim(&mut self, new_account_id: AccountId, new_public_key: String) -> Promise {
+    pub fn create_account_and_claim(&mut self, new_account_id: AccountId, new_public_key: Base58PublicKey) -> Promise {
```

Comment:
You can verify that the account is valid in this function. It's not required because the runtime going to roll it back anyway once the full execution is completed, but it would save some gas. See:
`env::is_valid_account_id(new_account_id.as_bytes())`